native/src/main/java/io/ballerina/stdlib/nats/basic/consumer/DefaultMessageHandler.java

```diff
@@ -393,43 +367,32 @@ public void notifySuccess(Object obj) {
         public void notifyFailure(BError error) {
             error.printStackTrace();
             natsMetricsReporter.reportConsumerError(subject, NatsObservabilityConstants.ERROR_TYPE_MSG_RECEIVED);
-            countDownLatch.countDown();
             // Service level `panic` is captured in this method.
             // Since, `panic` is due to a critical application bug or resource exhaustion
             // we need to exit the application.
             // Please refer: https://github.com/ballerina-platform/ballerina-standard-library/issues/2714
-            System.exit(1);
         }
     }
 
     /**
      * {@code NatsTypeCheckCallback} provides ability to check whether a given type is a subtype of
      * nats:AnydataMessage.
      */
-    public static class NatsTypeCheckCallback implements Callback {
-
-        private final Semaphore semaphore;
+    public static class NatsTypeCheckCallback implements Handler {
```

Comment:
Shall we rename this and `ResponseCallback` to `NatsTypeCheckHandler` and `ResponseHandler`? Because having callback suffix can be confusing.

Reply:
I Agree. Will make the change.

Reply:
Fixed in c5b69255471f5abbdfa255588b555a3c12d8d6c5